### ISSUE TITLE
Extracting the tar content of undefined failed

### ISSUE THREAD

**Issue by opiation** [yarnpkg/yarn#6312]
**Do you want to request a *feature* or report a *bug*?**
Reporting a bug when running `yarn install` to install node dependencies.  For severity, this bug seems **critical** considering it essentially prevents me from acquiring node dependencies.

**What is the current behavior?**
Fails *sometimes* with errors like the following:

```
yarn install v1.9.4
[1/4] Resolving packages...
[2/4] Fetching packages...
[1/4] Resolving packages...
[2/4] Fetching packages...
[1/4] Resolving packages...
[2/4] Fetching packages...
error https://registry.yarnpkg.com/lodash/-/lodash-4.17.10.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "ENOENT: no such file or directory, chmod '/usr/local/share/.cache/yarn/v2/npm-lodash-4.17.10-1b7793cf7259ea38fb3661d4d38b3260af8ae4e7/_cacheHas.js'"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

```
yarn install v1.9.4
[1/4] Resolving packages...
[2/4] Fetching packages...
[1/4] Resolving packages...
[2/4] Fetching packages...
[1/4] Resolving packages...
[2/4] Fetching packages...
error https://registry.yarnpkg.com/lodash/-/lodash-4.17.10.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "EEXIST: file already exists, mkdir '/usr/local/share/.cache/yarn/v2/npm-lodash-4.17.10-1b7793cf7259ea38fb3661d4d38b3260af8ae4e7'"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

```
yarn install v1.9.4
[1/4] Resolving packages...
[2/4] Fetching packages...
[1/4] Resolving packages...
[2/4] Fetching packages...
[1/4] Resolving packages...
[2/4] Fetching packages...
error https://registry.yarnpkg.com/fbjs/-/fbjs-0.8.17.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "ENOENT: no such file or directory, chmod '/usr/local/share/.cache/yarn/v2/npm-fbjs-0.8.17-c4d598ead6949112653d6588b01a5cdcd9f90fdd/lib/resolveImmediate.js'"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command
```

The occurrence of this error is the challenging part.  It does not always fail and does not always fail with the same dependency.  The installation is sometimes successful after 3-5 tries.

**If the current behavior is a bug, please provide the steps to reproduce.**
I've attempted to install dependencies on bare-metal and in a `node:8-alpine` docker container.  Both can *sometimes* encounter the error.  I've tested this on my personal device in Montreal, Canada (Mac OS X10.13), on a AWS EC2 instance (Ubuntu 18.04), on a GCE instance (Ubuntu 16.04) and on a production server in France (Debian 8).  Each of them can sometimes encounter this error.  I've also tried installing with and without  `yarn.lock` to no avail.  Find a `package.json` that appears to *sometimes* reproduce the issue [in this gist][0].  The issue does not seem to happen with projects having fewer dependencies.

**What is the expected behavior?**
Successful installation of all packages like `npm install` or `npm ci` which deterministically succeeds without any tar or caching errors.

**Please mention your node.js, yarn and operating system version.**
Tested with the following version:
Node: 8 LTS, 10
Yarn: 1.9.2, 1.9.4
OS: Ubuntu 18.04 LTS, Ubuntu 16.04 LTS, Debian 8, Mac OSX 10.13
Registrie: `registry.yarnpkg.com`, `registry.npmjs.org`, private registry

If you require any additional information, don't hesitate to request it.  FWIW, reducing the network-concurrency *seems* to produce a slightly higher success ratio but not consistently enough to conclude the errors are related.  It may be an area to investigate however.  I've unfortunately exhausted all the time I could afford to spend on this after a few days of troubleshooting.  I've reluctantly had to migrate all our CI builds back to using `npm install`/`npm ci` :(

[0]: https://gist.github.com/opiation/d6e4533990ddeeb83238218b3f6a11fe

**Comment by Titozzz** [comment]
Same issue, it's blocking my CI too, we updated recently to yarn 1.9.2

**Comment by adrienharnay** [comment]
@opiation The error is indeed random but I may have found the cause: do you have distant git URLs in your package.json without `.git` at the end? We had two of those and adding `.git` fixed the issue. Not sure why the error message doesn't directly state this is is the problem though.

**Comment by adrienharnay** [comment]
Also, maybe related: https://github.com/yarnpkg/yarn/issues/5536

**Comment by adrienharnay** [comment]
Distant wasn't the right word, I just meant packages installed from Git 🙂

Could you try this?

`"storybook-addon-markdown": "https://github.com/mihalik/storybook-addon-markdown.git"`

**Comment by opiation** [comment]
As per my previous comment, I still encounter the issue without `storybook-addon-markdown` dependency.  Thus, I don't believe this issue stems from yarn improperly handling git URLs.

**Comment by adrienharnay** [comment]
Indeed, I read too quickly. Well, that fixed our bug, but I have no idea about yours 😕 Sorry

**Comment by Titozzz** [comment]
@opiation Did you update you yarn.lock file too? Cause I had to do that

**Comment by opiation** [comment]
@Titozzz, I encounter this error with and without the `yarn.lock` file.  I have deleted and recreated the lock file several times to no avail.

**Comment by ekarlso** [comment]
I get this too and I dont have any packages from git.

**Comment by luwes** [comment]
I wanted to get around this issue (https://github.com/yarnpkg/yarn/issues/6256) by using the tarball versions of the packages but indeed the error above is thrown for tarball urls on self-hosted Github enterprise.

github.com hosted tarballs somehow did work. e.g.
https://github.com/luwes/chameleon/archive/grasshopper-v0.0.1-beta.4.tar.gz

**Comment by khendry** [comment]
I'm seeing the same issue with a project we have. However when I remove the deps that run a `prepare` script as part of the install (due to being git urls) then it works. These happen to be pointing git urls, but I think it's actually the `prepare` that seem to kick off more `yarn install` processes that seem to subvert the mutex flag for some reason. I wonder if that's because the other processes are kicked off by the root process rather than by different root processes.  I don't know if this information helps or if its actually way off base.  But I figured I would share what I've found regardless.

**Comment by adrienharnay** [comment]
@khendry I got the issue again, and you are right, it comes from git dependencies which have a `prepare` script in their package.json! :+1:

**Comment by bjentsch** [comment]
Any news on this? We're facing this issue too.

**Comment by DexterHD** [comment]
Same problem. It's impossible to use yarn with CI. Every 2nd build failed with this error 😞

**Comment by holyxiaoxin** [comment]
try delete node_modules, 
```
yarn cache clean
yarn install --network-concurrency 1
```

**Comment by muuki88** [comment]
Thanks for sharing this. It's at least a workaround 🤗 , but no real solution if you want your build time reasonable fast 😅

**Comment by khendry** [comment]
I spoke too soon. I had asked my team mate if we'd tried this, rather than actually trying myself and he was _very_ confident that we had...he was mistaken and miss-read the previous post thinking it was related to the mutex flag, not network concurrency.  We've since re-tried and can confirm this does seem to also address the issue for us.

**Comment by hulkish** [comment]
while it's helpful to have a temp fix, obviously setting `--network-concurrency 1` is not ideal, as this will definitely make installs slower.

Are there any leads on the actual cause of this, yet? I'd like to take a look.

**Comment by hulkish** [comment]
@arcanis @rally25rs more details on this error:

<img width="1920" alt="screen shot 2018-10-28 at 10 04 18 am" src="https://user-images.githubusercontent.com/139332/47616904-f845ac80-da98-11e8-85d9-ab700c508ed4.png">

**Comment by hulkish** [comment]
@opiation out of curiosity, can you paste your package.json? Are you using the following "override" technique anywhere?

```json
"dependencies": {
  "foo": "npm:@myorg/foo"
}
```

**Comment by opiation** [comment]
@hulkish, as per my initial post, [here's the gist][0] I created with my `package.json`, `yarn.lock` and all the tests I ran that all produced the error described.  To clarify, each line in `failing_test.sh` can encounter this error but not consistently.  They may have to be tried more than once to encounter the error.  Just to have it in this thread, I'll summarize each test below:

### Failing tests
- `yarn install`
- `yarn install --frozen-lockfile`
- `yarn install --pure-lockfile`
- `yarn install --mutex network`
- `yarn install --network-concurrency 1`
- All of the above tests with `rm yarn.lock` beforehand
- All of the above tests in a `node:alpine` container with `git` installed (alpine at the time this thread was created)
- All of the above tests in a `node:8-alpine` container with `git` installed

As for the "override" technique, I am uncertain what you mean.  If you are referring to the _protocol_-like prefix in the dependency value (like `npm:` in your example), then yes, one dev dependency uses a `github` package:

`"storybook-addon-markdown": "github:mihalik/storybook-addon-markdown"`

The errors are still encountered however even when I remove the dev dependency so this does not seem related.

[0]: https://gist.github.com/opiation/d6e4533990ddeeb83238218b3f6a11fe

**Comment by arcanis** [comment]
ping @imsnif ? It seems related to the integrity check, per @hulkish's comment

**Comment by Aminelahlou** [comment]
@khendry Not using prepare anymore on our git dependencies resolved this for our ci, while --network-concurrency 1,  --child-concurrency 1 and --skip-integrity-check were not enough

**Comment by schneidmaster** [comment]
We were able to fix this with `npm config set always-auth true` (as documented [here](https://docs.npmjs.com/misc/config#always-auth)). As best I can tell, npm by default will provide your credentials _only_ for publishing packages, not for fetching them. For some reason yarn previously didn’t respect that setting, but it now does.

**Comment by mvonballmo** [comment]
I recently had this issue, using `yarn 1.12.3` and `node 10.13.0`. After trying many of the solutions above, to no avail, deleting/regenerating the `yarn.lock` file worked.

**Comment by OmgImAlexis** [comment]
I've been getting a similar issue. Removing the `yarn.lock` like @mvonballmo suggested was the only thing that worked.

```
yarn install v1.12.3
[1/4] Resolving packages...
[2/4] Fetching packages...
error https://registry.yarnpkg.com/iconv-lite/-/iconv-lite-0.4.23.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "ENOSPC: no space left on device, write"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by imsnif** [comment]
Hi friends,

So, judging by the different errors reported here - this actually seems like it can potentially be several different issues:
`ENOSPC: no space left on device, write`, 
` wanted sha1-Sl7Hxk364iw6FBJNus3uhG2Ay8Q= but got sha1-AHoWKXweP+Pg9aZkGBsAjFruGaM=` (btw, takes a hard look, but these are not the same), 
`the file appears to be corrupt: "Unexpected end of data"`, etc.

While I appreciate these can be happening in a similar place, they might be caused by entirely different issues and/or environments. The integrity check (specifically the untarStream on error callback - thanks for the detailed debug @hulkish!) is a funnel that can gather many errors, and it's a little hard to give feedback beyond the actual error to the user.

The above is especially true for the integrity migration (populating an old-style `yarn.lock` with the new integrity fields), since this one-time process (one time assuming it succeeds ofc) is more network intensive than a normal install (it loops through all packages without an `integrity` field and fetches their registry manifest).

The theories about a race condition are interesting and very definitely a possibility, I'd be happy to look into them further. I'm afraid though that @opiation 's reproduction did not work for me. I'm now running my 7th local installation of it and it still works without issue (I did not run the script, rather I just ran `yarn` to install it with that package.json and yarn.lock - I understand this still caused the issue for you?)

@opiation - can you still reproduce this issue? Under the same conditions? Maybe we can go down a level of resolution and you can tell me everything you do down to the commands to make this happen?

Anyone else on this thread has a setup they can share that reproduces this issue even partially consistently? I'd be very happy to get to the bottom of this.

**Comment by JeremyTCD** [comment]
I encountered the same error message in my CI system: 

```
2018-11-12T04:32:13.0386630Z error https://pkgs.dev.azure.com/JeremyTCD/_packaging/Main/npm/registry/cheerio/-/cheerio-0.22.0.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "Unexpected end of data"
2018-11-12T04:32:20.4838361Z 
2018-11-12T04:32:20.4852626Z     yarn install v1.12.3                                                                    
2018-11-12T04:32:20.4853491Z     [1/4] Resolving packages...                                                             
2018-11-12T04:32:20.4855400Z     [2/4] Fetching packages...                                                              
2018-11-12T04:32:20.4856037Z     info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

Managed to figure it out my particular issue though. Thought I'd leave a note here for anyone who encounters something similar:

## Cause
I called `yarn install` on my local machine after adding a new dependency to my project (cheerio@0.22.0). Due to a local `.npmrc`, yarn restored the dependency from a private registry of mine. The generated [`yarn.lock`](https://github.com/JeremyTCD/Mimo/blob/e943881aa3e3a1af8c1b6ffde119cfa13f063f6c/yarn.lock) contained the following lines:

```
cheerio@0.22.0:
  version "0.22.0"
  resolved "https://pkgs.dev.azure.com/JeremyTCD/_packaging/Main/npm/registry/cheerio/-/cheerio-0.22.0.tgz#a9baa860a3f9b595a6b81b1a86873121ed3a269e"
  dependencies:
  ...
```

Note how the package was resolved from a private repository. On my CI machine, I didn't have a `.npmrc` with the credentials for the private registry. This was the cause of the error message: 
```
https://pkgs.dev.azure.com/JeremyTCD/_packaging/Main/npm/registry/cheerio/-/cheerio-0.22.0.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "Unexpected end of data"
```

I fixed my local `.npmrc` and regenerated my [`yarn.lock`](https://github.com/JeremyTCD/Mimo/blob/587393b8f5f762f9230065f597bb99729b22b28a/yarn.lock):
```
cheerio@0.22.0:
  version "0.22.0"
  resolved "http://registry.npmjs.org/cheerio/-/cheerio-0.22.0.tgz#a9baa860a3f9b595a6b81b1a86873121ed3a269e"
  integrity sha1-qbqoYKP5tZWmuBsahocxIe06Jp4=
```
Note how the package is now resolved from the default NPM registry. The error stopped ocurring once I did this.

## Fix
If the cause of your issue is the same as mine, you can:
- Add required credentials to your CI machine or
- Tweak your local `.npmrc` (`yarn config list` will print the registry that yarn restores from), then regenerate `yarn.lock`.

## Notes
Perhaps the error message could be more specific.

**Comment by gregorybolkenstijn** [comment]
TL;DR: try deleting your yarn.lock file and generating it again. 

I got an error when trying to build on Netlify: `Extracting tar content of undefined failed, the file appears to be corrupt: "Unexpected end of data"`

Deleting the node_modules folder and the yarn.lock file and then generating them again via `yarn install` got me a new yarn.lock file with different dependencies. With this new file Netlify succesfully built my project.

**Comment by oderby** [comment]
@imsnif agreed that there are seemingly multiple different issues being reported here. I believe I have a repro case from a project I'm working on that triggers the problem outlined by @khendry [here](https://github.com/yarnpkg/yarn/issues/6312#issuecomment-422806004)

> I'm seeing the same issue with a project we have. However when I remove the deps that run a `prepare` script as part of the install (due to being git urls) then it works. These happen to be pointing git urls, but I think it's actually the `prepare` that seem to kick off more `yarn install` processes that seem to subvert the mutex flag for some reason. I wonder if that's because the other processes are kicked off by the root process rather than by different root processes.

Sharing repro steps below in the hopes it allows you to reproduce the problem. Let me know if you need more information.

### Repro steps
1. With node `v10.3.0` and yarn `v1.12.3`, in a new test folder, download the `package.json` and `yarn.lock` from [this gist](https://gist.github.com/oderby/95f5c59a3088e8eb52402818d5cba62d)
2. run `rm -rf ~/.cache/yarn* node_modules/ && yarn install --frozen-lockfile --network-concurrency 16` (clear cache and previously install node modules for reliable environment. Set concurrency high to increase chances of hitting issue)
3. observe output like the following:
```bash
yarn install v1.12.3
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
warning Pattern ["object-assign@latest"] is trying to unpack in the same destination "/home/ocderby/.cache/yarn/v4/npm-object-assign-4.1.1-2109adc7965887cfc05cbbd442cac8bfbb360863/node_modules/object-assign" as pattern ["object-assign@^4.1.0","object-assign@^4.1.0","object-assign@^4","object-assign@^4.1.1","object-assign@^4.1.0","object-assign@4.x","object-assign@^4.1.0","object-assign@^4.1.1","object-assign@^4.1.0","object-assign@^4.1.0","object-assign@^4.1.0","object-assign@^4.1.0","object-assign@^4.1.0","object-assign@^4.1.0","object-assign@^4.1.0","object-assign@^4.1.1","object-assign@^4.1.1","object-assign@^4.0.1","object-assign@^4.0.1","object-assign@^4.1.0","object-assign@^4.0.1","object-assign@^4.0.1","object-assign@^4.0.1","object-assign@^4.1.0","object-assign@^4.0.1"]. This could result in non-deterministic behavior, skipping.
info No lockfile found.
[1/4] Resolving packages...
warning eslint > file-entry-cache > flat-cache > circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
warning jest > jest-cli > prompts > kleur@2.0.2: Please upgrade to kleur@3 or migrate to 'ansi-colors' if you prefer the old syntax. Visit <https://github.com/lukeed/kleur/releases/tag/v3.0.0\> for migration path(s).
[2/4] Fetching packages...
error https://registry.yarnpkg.com/lodash/-/lodash-4.17.4.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "ENOENT: no such file or directory, chmod '/home/ocderby/.cache/yarn/v4/npm-lodash-4.17.4-78203a4d1c328ae1d86dca6460e369b57f4055ae/node_modules/lodash/_shortOut.js'"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

### Other notes
I've tried a variety of things, here are my notes:
1. The issue does not reproduce 100% of the time for me. As noted above, increasing the network concurrency used seems to make the issue more likely to occur.
2. Using a version of `react-textarea-autosize` published on the package registry makes the problem go away (seeming to confirm what @khendry reported above)
3. Setting `--mutex file` doesn't seem to help here at all
4. As reported above, if I limit network concurrency to `1` (via `--network-concurrency 1` argument), everything installs properly, albeit slower.
5. I've reproduced this on node v8.12.0, with both yarn v1.9.4 and v1.12.3. This was running on the `circleci/node:8-stretch` docker image running on Circle CI 2.0.

**Comment by ankurk91** [comment]
I start seeing this error recently after updating yarn to `1.12.3`.
See my travis-ci build failure https://travis-ci.org/ankurk91/vue-cleave-component

```
$ yarn install --non-interactive
yarn install v1.12.3
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
error https://registry.yarnpkg.com/har-validator/-/har-validator-5.1.2.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "Unexpected end of data"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
The command "yarn install --non-interactive" failed and exited with 1 during .
```
It is happening with only with `har-validator@5.1.2`.
I will post back if i got success somehow.

**Comment by garaboncias** [comment]
I also get 
```error https://registry.yarnpkg.com/har-validator/-/har-validator-5.1.2.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "Unexpected end of data"```
with curl i got 404 for https://registry.yarnpkg.com/har-validator/-/har-validator-5.1.2.tgz
but in my browser i can download it.
One of my server if i downgrade yarn to 1.12.1 it start to work but on the other server even if i downgrade the error remains the same.
Is that possible that is some kind of cloudflare (config) issue?

**Comment by webmobiles** [comment]
same error here, 
easy to reproduce:
`
yarn upgrade typescript@^2.8`

then:
`yarn upgrade typescript@2.8`

I did ctrl+c while installing this last package.. and when I try 'yarn upgrade' again  I get:
```

yarn upgrade v1.12.3
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error https://registry.yarnpkg.com/typescript/-/typescript-2.8.4.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "ENOENT: no such file or directory, stat '/Users/u/Library/Caches/Yarn/v4/npm-typescript-2.8.4-0b1db68e6bdfb0b767fa2ab642136a35b059b199/node_modules/typescript/lib/lib.d.ts'"
info Visit https://yarnpkg.com/en/docs/cli/upgrade for documentation about this command.

```

**Comment by csvan** [comment]
Seeing this for multiple packages in our CI environment:

```
responsive-props-1.2.2.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "Invalid tar header. Maybe the tar is corrupted or it needs to be gunzipped?"
```

```
styled-components-breakpoint-2.1.3.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "Invalid tar header. Maybe the tar is corrupted or it needs to be gunzipped?"
```

**Comment by jbergens** [comment]
I had a similar problem but I get ....the file appears to be corrupt:  "EBUSY: ...". 
I cleared my whole yarn cache an re-run and still got the same error so it seems like yarn is creating files and locking them for itself. 

This is on Windows 10.

`yarn install v1.10.1
[1/4] Resolving packages...
[2/4] Fetching packages...
error https://registry.yarnpkg.com/fbjs/-/fbjs-0.8.17.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "EBUSY: resource busy or locked, open 'c:\\src\\yarn\\cache\\v2\\npm-fbjs-0.8.17-c4d598ead6949112653d6588b01a5cdcd9f90fdd\\lib\\UserAgent.js'"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.`

**Comment by davidalee** [comment]
Removing `yarn.lock` was what got it working for me.

**Comment by nothingismagick** [comment]
@binchik - this is the only thing that worked for me.

**Comment by nothingismagick** [comment]
So after rewinding back to the series of events leading up to `yarn` failing, I believe I executed an npm script in a package.json that was something like this:

```json
"audit": "npm audit"
```

Which is totally silly, because I never use `npm` in that project. After this command everything (including npm) would just start to have random faults and never complete, in line with the experience of others in this thread.

**Comment by deleteme** [comment]
FWIW, I encountered this problem today. A few things:

* Removing `.npmrc` fixed it. The only thing in the file had to do with the auth token.
* `npm install` failed too, and it logged a 401 unauthorized error.
* After removing the `.npmrc` file, `npm install` worked again.

**Comment by hulkish** [comment]
@deleteme per my findings, this sounds more like a byproduct of the bug, rather than the cause.

**Comment by arcanis** [comment]
Given that this problem suddenly appears much more than usual and that it's during [the npm registry being especially flaky](https://npm.community/t/can-t-install-scoped-packages-e401-unauthorized/3631/15), it's quite likely my hypothesis isn't far off

**Comment by mjniday** [comment]
@arcanis just started having this problem today. I can confirm that by removing that npmrc auth token line the issue was resolved.

**Comment by achillesrasquinha** [comment]
Facing the issue. There's no '~/.npmrc' created in my case.

**Comment by chris-snowden** [comment]
As @achillesrasquinha said, deleting yarn.lock and running yarn again to regenerate yarn.lock fixed the problem for me. One of the package versions had incremented in package.json but the updated yarn.lock file wasn't checked in.

**Comment by JeongUkJae** [comment]
In my case, I use CircleCI, `circleci/node:10.11.0` docker image, `yarn@1.12.3`, and there is no `~/.npmrc`. Thank you @achillesrasquinha. It works for me.

**Comment by sanu-explore** [comment]
I have been facing this issue over a week. `yarn install --network-concurrency 1 ` solved issue but it is very very slow. 

In btw, this info could be helpful to anyone. 
 I was using a custom npm package (in house) in my project. Always I am getting same issue like `.cache/v4` but showing different package names every fail. After spending much time, I find one random observation. 
My project and custom npm package is using same `yarn build` for build the bundle. I have updated my custom package build script name to another name as `yarn build:p`. Then it is start working. I ran build many times. It was not failed. Not sure How these 2 are dependent but worked for me.

**Comment by 0x80** [comment]
Removing `.npmrc` only didn't do it for me. I also had to remove my `yarn.lock` file like @davidalee mentioned. I don't know why he's getting thumbs down for it 🤷‍♂️

**Comment by zedtux** [comment]
I'm not really fan of deleting the `yarn.lock` file so what I did is just remove the `har-validator` package from the `yarn.lock` and then re-run `yarn` which fixed the issue for me.

**Comment by phattruong-agilityio** [comment]
Yes `rm yarn.lock` work for me. Fating issue with package `har-validator-5.1.2`. 
```
error https://registry.yarnpkg.com/har-validator/-/har-validator-5.1.2.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "Unexpected end of data"
```

**Comment by ondratra** [comment]
Hi, har-validator-5.1.2 was unpublished from npm as stated here https://github.com/ahmadnassri/node-har-validator/issues/112#issuecomment-437378269 , thus you need to upgrade your dependencies via `yarn upgrade`(this has probably the same effect as removing `yarn.lock` that was recommended by the others).

I suppose this issue can be closed.

**Comment by arcanis** [comment]
The problem is that the npm registry is generally unstable and returns errors (at a higher rate when multiple requests are firing apparently - maybe some kind of per-ip throttling?). For some reason they are not properly caught by Yarn, which blindly tries to hash them and compare them to the expected hash - which fails.

So there is a bug in Yarn (we should print a more helpful error), but given that the real problem is how flaky the npm registry is, it's not my priority at the moment (I'd definitely review a PR, though!).

As for why it doesn't happen with npm: they retry their requests until they work. Yarn has a mechanism to do that, but not on the part that specifically compute the hash.

I'd suggest to use an [offline mirror](https://yarnpkg.com/blog/2016/11/24/offline-mirror/) to stop relying on the npm registry for your installs.

**Comment by jaychoumylove** [comment]
> There's no `~/.npmrc` created in my case. But regenerating `yarn.lock` worked for me.
> 
> Simply,
> 
> ```
> $ rm yarn.lock && yarn
> ```

thank you, 
`rm -rf ./yarn.lock && yarn`
it's work for me!

**Comment by mrseanryan** [comment]
in case it helps:

- this same error occured for me, when I had forgotten to log in to npm (doh!)

**Comment by middlehut** [comment]
For me the issue was resolved with `service docker restart` (Ubuntu 18.04).

**Comment by jleppert** [comment]
I've been experiencing intermittent and non-deterministic errors like this one. I restart my build, nothing else having changed and it works. Does anyone have any alternatives to yarn?

**Comment by quarklemotion** [comment]
I starting getting this same error on every build (errors on a different npm module each time) after making a PR to update our base docker image from `node:8.12.0` to `node:8.13.0`.  I inspected these node docker images and discovered that the pre-installed yarn version was changed from `v1.9.4` to `v1.12.3`.  See: [associated git commit](https://github.com/nodejs/docker-node/commit/0ceefee0745a078709c60d92e4229f5eb6a3df42#diff-c27c14abc33ff0f1ffaf5aaa5b6bd6eaR49).  I tried some of the suggested fixes in this thread with no luck in resolving the error.  I was able to fix the issue by simply [downgrading the yarn version](https://github.com/nodejs/docker-node/blob/master/docs/BestPractices.md#upgradingdowngrading-yarn) in my Dockerfile to `v1.9.4`.  I know this version of yarn has been problematic for others, but for me the more recent yarn version is triggering the issue.  I'll note that I am using an `.npmrc` file that provides credentials to access private modules via jfrog artifactory.

**Comment by kunokdev** [comment]
Why is still closed? Still breaking CI

**Comment by arcanis** [comment]
> In the meantime I'll close this discussion, as the error messages will change and this thread grown quite large (we can open new ones to discuss each status code individually).

I'll go ahead and lock this thread since it seems to me it reached past its usefulness. As a reminder:

- If you have this error message, you are **very** likely using an old release. Upgrade to 1.13+ to get the *true* error message. Chances are the registry is returning an HTTP 500 for some reason.

- If you still get errors that seem to come from Yarn itself open a new thread and detail how to reproduce the problem. If you don't provide a reproduction we won't be able to provide a fix, and will likely have to ask you to investigate yourself.
